OSDOCS-3700: Technology preview release notes for heterogeneous support (#46154)
openshift/openshift-docs: release_notes/ocp-4-11-release-notes.adoc

[id="ocp-4-11-worker-nodes"]
==== Default AWS and VMware vSphere compute node resources
Beginning with {product-title} 4.11, by default, the installation program now deploys AWS and VMware vSphere compute nodes with 4 vCPUs and 16 GB of virtual RAM

[id="ocp-4-11-heterogeneous-tech-preview"]
==== {product-title} on Heterogeneous architectures (Technology Preview)
{product-title} 4.11 introduces Heterogeneous architecture cluster support using Azure and AWS installer-provisioned infrastructure in Technology preview.

> This feature offers the ability to deploy a cluster with `x86_64` control planes and mixed `x86_64`/`arm64` worker nodes exclusively for Azure and AWS instances.

Reviewer: would be good to call out that adding workers of arm64 architecture is a day2 operation

Author: I'll look into that, thank you!

For more information, see xref:../installing/installing_bare_metal_ipi/ipi-install-installation-workflow.adoc#configuring-the-bios_ipi-install-installation-workflow[Configuring the BIOS] and xref:../installing/installing_bare_metal_ipi/ipi-install-installation-workflow.adoc#configuring-the-raid_ipi-install-installation-workflow[Configuring the RAID].

[id="ocp-4-11-heterogeneous-tech-preview"]
==== {product-title} on Heterogeneous architectures (Technology Preview)
{product-title} 4.11 introduces Heterogeneous architecture cluster support using Azure installer-provisioned infrastructures in Technology preview.

> This feature offers the ability to add  `arm64` or `x86_64`  worker nodes to an existing Azure cluster with a `x86_64` control plane as a day two operation. Control planes on `arm64` architectures are not currently supported.

Reviewer: i think we should  make it clear here that the cluster should be provisioned as a "single" arch cluster (homogeneous) using the normal IPI process but with an installer made available here: https://mirror.openshift.com/pub/openshift-v4/multi/clients/ocp-dev-preview/ 

once that is done, worker nodes of any architecture can be added.

Author: @Prashanth684 Hi Prashanth! Just a question, so the Azure IPI install for heterogeneous is the same, except that when you deploy the cluster, you must override the release image and use the mixed-arch installer that you added above?

Reviewer: if you use the installer binary from the mirror link i posted above, you should not need to override the image. it will  install with the correct heterogeneous payload.

For more information, see the "Installation configuration parameters" section of the installation documentation for your platform.

[id="ocp-4-11-heterogeneous-tech-preview"]
==== {product-title} on Heterogeneous architectures (Technology Preview)
{product-title} 4.11 introduces Heterogeneous architecture cluster support using Azure installer-provisioned infrastructures in Technology preview.

> This feature offers the ability to add  `arm64` worker nodes to an existing Azure installer-provisioned cluster with a heterogeneous installer binary as a day two operation. You can add `arm64` workers to your heterogeneous cluster by creating a custom Azure machine set that uses a manually generated `arm64` bootimage. Control planes on `arm64` architectures are not currently supported. For more information, see xref:../post_installation_configuration/deploy-heterogeneous-configuration.adoc[Configuring a heterogeneous cluster].

Reviewer: maybe call out explicitly that the provisioned cluster must be x86 before adding the workers.

> This feature offers the ability to add  `arm64` worker nodes to an existing Azure  installer-provisioned `x86_64` cluster with a heterogeneous installer binary as a day two operation. You can add `arm64` workers to your heterogeneous cluster by creating a custom Azure machine set that uses a manually generated `arm64` bootimage. Control planes on `arm64` architectures are not currently supported. For more information, see xref:../post_installation_configuration/deploy-heterogeneous-configuration.adoc[Configuring a heterogeneous cluster].

Reviewer: sorry another nit: 
```
offers as a day two operation, the ability to add  `arm64` worker nodes to an existing Azure `x86_64` cluster, installer provisioned with a heterogeneous installer binary
```
?

For more information, see xref:../post_installation_configuration/cluster-capabilities.adoc#cluster-capabilities[Cluster capabilities].

[id="ocp-4-11-heterogeneous-tech-preview"]
==== {product-title} on heterogeneous architectures (Technology Preview)

> {product-title} 4.11 introduces heterogeneous architecture cluster support using Azure installer-provisioned infrastructure in Technology preview. This feature offers, as a day-two operation, the ability to add `arm64` worker nodes to an existing `x86_64` Azure cluster that is installer provisioned with a heterogeneous installer binary. You can add `arm64` workers to your heterogeneous cluster by creating a custom Azure machine set that uses a manually generated `arm64` bootimage. Control planes on `arm64` architectures are not currently supported. For more information, see xref:../post_installation_configuration/deploy-heterogeneous-configuration.adoc[Configuring a heterogeneous cluster].

Reviewer 2: ```suggestion
{product-title} 4.11 introduces heterogeneous architecture cluster support using Azure installer-provisioned infrastructure in Technology Preview. This feature offers, as a day-two operation, the ability to add `arm64` worker nodes to an existing `x86_64` Azure cluster that is installer provisioned with a heterogeneous installer binary. You can add `arm64` workers to your heterogeneous cluster by creating a custom Azure machine set that uses a manually generated `arm64` boot image. Control planes on `arm64` architectures are not currently supported. For more information, see xref:../post_installation_configuration/deploy-heterogeneous-configuration.adoc[Configuring a heterogeneous cluster].
```

> {product-title} 4.11 introduces heterogeneous architecture cluster support using Azure installer-provisioned infrastructures in Technology preview. This feature offers, as a day two operation, the ability to add `arm64` worker nodes to an existing `x86_64` Azure cluster that is installer provisioned with a heterogeneous installer binary. You can add `arm64` workers to your heterogeneous cluster by creating a custom Azure machine set that uses a manually generated `arm64` bootimage. Control planes on `arm64` architectures are not currently supported. For more information, see xref:../post_installation_configuration/deploy-heterogeneous-configuration.adoc[Configuring a heterogeneous cluster].

Reviewer 3: ```suggestion
{product-title} 4.11 introduces heterogeneous architecture cluster support using Azure installer-provisioned infrastructure in Technology preview. This feature offers, as a day two operation, the ability to add `arm64` worker nodes to an existing `x86_64` Azure cluster that is installer provisioned with a heterogeneous installer binary. You can add `arm64` workers to your heterogeneous cluster by creating a custom Azure machine set that uses a manually generated `arm64` bootimage. Control planes on `arm64` architectures are not currently supported. For more information, see xref:../post_installation_configuration/deploy-heterogeneous-configuration.adoc[Configuring a heterogeneous cluster].
```

Reviewer 3: @kelbrown20 take a look at this for the previous suggestion: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/term_glossary.adoc#installer-provisioned-infrastructure

> {product-title} 4.11 introduces heterogeneous architecture cluster support using Azure installer-provisioned infrastructures in Technology preview. This feature offers, as a day two operation, the ability to add `arm64` worker nodes to an existing `x86_64` Azure cluster that is installer provisioned with a heterogeneous installer binary. You can add `arm64` workers to your heterogeneous cluster by creating a custom Azure machine set that uses a manually generated `arm64` bootimage. Control planes on `arm64` architectures are not currently supported. For more information, see xref:../post_installation_configuration/deploy-heterogeneous-configuration.adoc[Configuring a heterogeneous cluster].

Reviewer 3: ```suggestion
{product-title} 4.11 introduces heterogeneous architecture cluster support using Azure installer-provisioned infrastructures in Technology preview. This feature offers, as a day-two operation, the ability to add `arm64` worker nodes to an existing `x86_64` Azure cluster that is installer provisioned with a heterogeneous installer binary. You can add `arm64` workers to your heterogeneous cluster by creating a custom Azure machine set that uses a manually generated `arm64` bootimage. Control planes on `arm64` architectures are not currently supported. For more information, see xref:../post_installation_configuration/deploy-heterogeneous-configuration.adoc[Configuring a heterogeneous cluster].
```
